server/src/test/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsActionTests.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.rest.FakeRestRequest;
+
+import java.util.HashMap;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.object.HasToString.hasToString;
+import static org.mockito.Mockito.mock;
+
+public class RestClusterStatsActionTests extends OpenSearchTestCase {
+
+    private RestClusterStatsAction action;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        action = new RestClusterStatsAction();
+    }
+
+    public void testUnrecognizedMetric() {
+        final HashMap<String, String> params = new HashMap<>();
+        final String metric = randomAlphaOfLength(64);
+        params.put("metric", metric);
+        final RestRequest request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_cluster/stats").withParams(params).build();
```

Comment:
Can we have the metric and index_metric as part of Path and not Params ?

Reply:
FakeRestRequest doesn't have support to compute the params from URI path. Responsibility is with the clients to set params.